src/codegen/ruby/def/field.rb

```diff
@@ -27,7 +27,8 @@ def java_name
   def java_type
     jt = get_java_type type.tr('?', '')
     
-    return "java.util.Optional<#{jt}>" if optional?
+    jt = "com.github.princesslana.eriscasper.data.util.Nullable<#{jt}>" if nullable?
+    jt = "java.util.Optional<#{jt}>" if optional?
```

Comment:
What if it's nullable and optional? This looks like an issue.

Reply:
I don't think having an Optional<Nullable<X>> is a very good idea. Which is what I think it will do at the moment.

Reply:
Yes, it will currently do `Optional<Nullable<T>>`. What would we gain by adding a `OptionalAndNullable<T>` type?

Reply:
Nesting generics is really ugly, imagine having Optional<Nullable<ImmutableList<Integer>>>>.
It's much rather see something less nested so I don't have to check and get to actually start accessing Nullable methods. It's much better imo to have access to methods which are used for what it is. If it's an Optional and Nullable type I'd much rather see access to Optional and Nullable methods. Makes more sense in my mind, I could be wrong about it. That's just my 2 cents about it.

Reply:
So, do we also have OptionalAndNullableImmutableList? OptionalImmutableList? NullableImmutableList?

Reply:
In my mind it works more of a `Identifier -> Container -> type`, the class is nullable, and optional, contained in a list, of type Integer, or the class is Optional, raw, and of type Boolean so `Optional<Boolean>` I don't think it's necessary to make every single possible outcome, however when it's the same identifier I don't find it good to nest them.

Reply:
It might be useful to make Optional and Nullable interfaces to throw in just so you can apply an Optional as a NullableAndOptional and a Nullable the same way. Might be more intuitive.

Reply:
I'm against making Optional and Nullable interfaces. There's already Java's optional, Guava's Optional, rxjava's Maybe. I don't think adding another optional type helps.

I'm not convinced the OptionalAndNullable type helps, because when something is Optional and Nullable it's because you have to treat null and empty differently.

So, you'll end up with
```java
if (thing.isNull()) {

} else if (!thing.isPresent()) {

} else {

}
```

My gut feel is that combing rather than nesting doesn't help this kind of code get any clearer (especially if using the map/flatMap/etc type methods).

I think the case of `Optional<Nullable<ImmutableList<T>>>` is not worth worrying about. I'm not sure this appears anywhere.

Reply:
Fair, maybe see how it's used later on and decide then if we want it changed. I don't think it's too much worth worrying about but it's something to keep in mind.